telemetry/code/models/p4-dtel-metadata-semantics.yang

```diff
@@ -0,0 +1,326 @@
+  }
+
+  identity METADATA_SIZE_UNITS {
+    base METADATA_UNITS;
+    description
+      "Base identity for specifying units of size, e.g. queue size";
+  }
+
+  identity METADATA_BYTES {
+    base METADATA_SIZE_UNITS;
+    description
+      "Encoding size in units of bytes";
+  }
+
+  identity METADATA_CELLS {
+    base METADATA_SIZE_UNITS;
+    description
+      "Encoding size in units of cells";
+  }
+
+  identity METADATA_PACKETS {
+    base METADATA_SIZE_UNITS;
+    description
+      "Encoding size in units of packets";
+  }
+
+  identity METADATA_UTILIZATION_UNITS {
+    base METADATA_UNITS;
+    description
+      "Base identity for specifying units of utilization";
+  }
+
+  identity METADATA_PERCENT {
+    base METADATA_UTILIZATION_UNITS;
+    description
+      "Encoding utilization in units of percent";
+  }
+
+  // typedef statements
+
+  typedef queue-md-timing {
+    type enumeration {
+      enum ENQUEUE_TIME {
+        description
+          "Retrieval of metadata at packet enqueue time";
+      }
+      enum DEQUEUE_TIME {
+        description
+          "Retrieval of metadata at packet dequeue time";
+      }
+    }
+    description
+      "Specifies the timing of retrieval of queue related metadata";
+  }
+
+  typedef int-last-hop-md-report-position {
+    type enumeration {
+      enum EMBEDDED_INT {
+        description
+          "Last hop metadata is included with embedded INT metadata";
+      }
+      enum SAME_TELEMETRY_REPORT {
+        description
+          "Last hop metadata is included in the Telemetry Report
+           header, in the same telemetry report as the embedded INT
+           metadata from previous hops";
+      }
+      enum SEPARATE_TELEMETRY_REPORT {
+        description
+          "Last hop metadata is included in the Telemetry Report
+           header, in a separate telemetry report from the embedded
+           INT metadata from previous hops";
+      }
+    }
+    description
+      "Specifies whether last hop metadata is included in the same
+       telemetry report as the embedded INT metadata from previous
+       hops, and whether the last hop metadata is included in
+       embedded INT metadata or in the Telemetry Report header";
+  }
+
+  // grouping statemenets
+
+  grouping dtel-md-semantics-time-units {
+    description
+      "Units of time used for specific instances of INT metadata and
+       telemetry report metadata";
+
+    leaf time-units {
+      type identityref {
+        base METADATA_TIME_UNITS;
+      }
+      description
+        "Units of time";
+    }
+  }
+
+  grouping dtel-md-semantics-size-units {
+    description
+      "Units of size used for specific instances of INT metadata and
+       telemetry report metadata";
+
+    leaf size-units {
+      type identityref {
+        base METADATA_SIZE_UNITS;
+      }
+      description
+        "Units of size";
+    }
+
+    leaf cell-size {
+      when "../size-units = 'METADATA_CELLS'" {
```

Comment:
Shouldn't this 'when' be around the entire leaf? I'm not sure why you'd make the description conditional, especially when there is another description below.

Reply:
From RFC 6020:
   The "when" statement makes its parent data definition statement
   conditional.  The node defined by the parent data definition
   statement is only valid when the condition specified by the "when"
   statement is satisfied.

Reply:
Got it, thanks. My yang is rusty!